Fix / work around the bug where 'ssh -o ControlMaster=auto'
interacts badly with the svn+ssh tunnel scheme.  As I note in
the bug log for #413102, I don't really think the problem is
either svn's or openssh's fault, but the obvious way to avoid
it is to avoid accidentally becoming a ControlMaster.  In this
mode we will still use ssh connection sharing, but only if a
ControlMaster is already present.

Index: subversion-1.7.0/subversion/libsvn_subr/config_file.c
===================================================================
--- subversion-1.7.0.orig/subversion/libsvn_subr/config_file.c	2011-10-20 20:47:32.000000000 +0200
+++ subversion-1.7.0/subversion/libsvn_subr/config_file.c	2011-10-20 21:55:52.000000000 +0200
@@ -1027,7 +1027,7 @@
         "### passed to the tunnel agent as <user>@<hostname>.)  If the"      NL
         "### built-in ssh scheme were not predefined, it could be defined"   NL
         "### as:"                                                            NL
-        "# ssh = $SVN_SSH ssh -q"                                            NL
+        "# ssh = $SVN_SSH ssh -q -o ControlMaster=no"                        NL
         "### If you wanted to define a new 'rsh' scheme, to be used with"    NL
         "### 'svn+rsh:' URLs, you could do so as follows:"                   NL
         "# rsh = rsh"                                                        NL
Index: subversion-1.7.0/subversion/libsvn_ra_svn/client.c
===================================================================
--- subversion-1.7.0.orig/subversion/libsvn_ra_svn/client.c	2011-10-20 20:47:31.000000000 +0200
+++ subversion-1.7.0/subversion/libsvn_ra_svn/client.c	2011-10-20 21:55:52.000000000 +0200
@@ -391,7 +391,7 @@
        * versions have it too. If the user is using some other ssh
        * implementation that doesn't accept it, they can override it
        * in the [tunnels] section of the config. */
-      val = "$SVN_SSH ssh -q";
+      val = "$SVN_SSH ssh -q -o ControlMaster=no";
     }
 
   if (!val || !*val)
